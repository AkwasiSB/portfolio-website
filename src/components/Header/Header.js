import PropTypes from 'prop-types';
import React from 'react';
import styles from './Header.scss';

export default class Header extends React.Component {
	static propTypes = {
		isSticky: PropTypes.bool.isRequired,
		activeNavLink: PropTypes.string.isRequired,
	}

	state = {
		isHomeActive: true,
		isAboutActive: false,
		isPortfolioActive: false,
	}

	componentDidUpdate(prevProps) {
		if (prevProps.activeNavLink !== this.props.activeNavLink) {
			this.setState({
				isHomeActive: this.props.activeNavLink === 'Home' ? true : false,
				isAboutActive: this.props.activeNavLink === 'About' ? true : false,
				isPortfolioActive: this.props.activeNavLink === 'Portfolio' ? true : false
			});
		}
	}

	onNavLinkClicked = (key) => () => {
		this.setState({
			isHomeActive: key === 'Home' ? true : false,
			isAboutActive: key === 'About' ? true : false,
			isPortfolioActive: key === 'Portfolio' ? true : false
		});
	}

	render() {
		const { isHomeActive, isAboutActive, isPortfolioActive } = this.state;
		const { isSticky } = this.props;

		return (
			<nav id='navbar' className={`${styles.navbar} ${isSticky ? styles.sticky : ''}`}>
				<a className={styles.webLogo} href='#'>
					<span className={styles.logoContent}>A</span>
				</a>

				<ul className={styles.navMenu}>
					<li>
						<a
							className={`${styles.navlink} ${isHomeActive ? 'active' : ''}`}
							href='#'
							onClick={this.onNavLinkClicked('Home')}>
							HOME
						</a>
					</li>

					<li>
						<a
							className={`${styles.navlink} ${isAboutActive ? 'active' : ''}`}
							href='#about'
							onClick={this.onNavLinkClicked('About')}>
							ABOUT
						</a>
					</li>

					<li>
						<a
							className={`${styles.navlink} ${isPortfolioActive ? 'active' : ''}`}
							href='#portfolio'
							onClick={this.onNavLinkClicked('Portfolio')}>
							PORTFOLIO
						</a>
					</li>
				</ul>
			</nav>
		);
	}
}
