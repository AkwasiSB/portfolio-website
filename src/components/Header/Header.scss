$screen-sm: 768px;
$screen-md: 1200px;

.navbar {
	position: relative;
	grid-area: navbar;
	background-color: #fff;
	display: flex;
	flex-direction: row;
	justify-content: center;
	align-items: center;
	padding: 0;

	@media (min-width: $screen-sm) {
		justify-content: space-between;
		padding: 0 10rem;
	}

	@media (min-width: $screen-md) {
		padding: 0 12.8rem;
	}

	&.sticky {
		position: fixed;
		top: 0;
		left: 0;
		right: 0;
		height: 6rem;
		padding: 0;
		z-index: 100;
		box-shadow: 0px 2px 4px 1px #eee;

		@media (min-width: $screen-sm) {
			padding: 0 10rem;
		}

		@media (min-width: $screen-md) {
			padding: 0 12.8rem;
		}
	}
}

.web-logo {
	position: absolute;
	bottom: -67%;
	left: 50%;
	transform: translateX(-50%);
	width: 4rem * 2;
	height: 4rem;
	border-bottom-left-radius: 4rem * 2;
	border-bottom-right-radius: 4rem * 2;
	background-color: rgba(128, 0, 128, 0.8);
	text-align: center;
	line-height: 4rem;
	margin-left: 0rem;
	z-index: 999;

	@media (min-width: $screen-sm) {
		position: relative;
		left: 0;
		bottom: 0;
		width: 5rem;
		height: 5rem;
		border-radius: 50%;
		line-height: 5rem;
		background-color: rgba(228, 101, 114, 0.85);
	}
}

.navbar.sticky .web-logo {
	box-shadow: 0 1px 1px 1px #888;

	@media (min-width: $screen-sm) {
		box-shadow: none;
	}
}

.logo-content {
	display: block;
	font-size: 2.8rem;
	font-weight: 600;
	color: #fff;
	
	@media (min-width: $screen-sm) {
		animation: Revolver 10s linear 2s infinite;
	}
}

.nav-menu {
	list-style: none;
	padding: 0;
	margin: 0;
	
	li {
		text-align: center;
		display: inline-block;
		padding: 1.2rem 0;
		width: 12rem;
		
		@media (min-width: $screen-sm) {
			width: auto;
		}
	}
}

:local(.navlink) {
	color: #999;
	font-size: 1.4rem;
	font-weight: 600;
	padding: 2rem;

	&:hover {
		color: rgb(119, 83, 119);
	}

	:global &.active {
		color: #000;
	}
}

@keyframes Revolver {
	from {
		transform: rotateX(0deg);
	}
	20% {
		transform: rotateX(90deg);
	}
	40% {
		transform: rotateX(180deg);
	}
	60% {
		transform: rotateY(0deg);
	}
	80% {
		transform: rotateY(90deg);
	}
	100% {
		transform: rotateY(180deg);
	}
}
