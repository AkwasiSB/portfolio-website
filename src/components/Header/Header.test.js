import Header from './Header';

it('should render <Header /> correctly', () => {
  const wrapper = shallow(<Header isSticky={false} activeNavLink='Home' />);

  expect(wrapper).toMatchSnapshot();
})