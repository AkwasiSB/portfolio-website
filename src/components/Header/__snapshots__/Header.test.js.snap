// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <Header /> correctly 1`] = `
<nav
  className="navbar "
  id="navbar"
>
  <a
    className="webLogo"
    href="#"
  >
    <span
      className="logoContent"
    >
      A
    </span>
  </a>
  <ul
    className="navMenu"
  >
    <li>
      <a
        className="navlink active"
        href="#"
        onClick={[Function]}
      >
        HOME
      </a>
    </li>
    <li>
      <a
        className="navlink "
        href="#about"
        onClick={[Function]}
      >
        ABOUT
      </a>
    </li>
    <li>
      <a
        className="navlink "
        href="#portfolio"
        onClick={[Function]}
      >
        PORTFOLIO
      </a>
    </li>
  </ul>
</nav>
`;
